docs/proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,319 @@
+
+Currently, cAdvisor collects a large number of metrics related to system and container performance. However, only some of these metrics are consumed by the kubelet summary API, and many are not used.  The kubelet summary API is published to the kubelet summary API endpoint.  Some of the metrics provided by the summary API are consumed internally, but most are not used internally.
+
+### Motivations
+
+Giving the kubelet the role of both providing metrics for its own use, and providing metrics for users has a couple problems
+ - First, it is clearly inefficent to collect metrics and not use them.  The kubelet uses only a small portion of the metrics it collects, and thus presents considerable extra overhead to any users who do not use them, or prefer a third party monitoring solution.
+ - Second, as the number of metrics collected grows over time, the kubelet will gain more and more overhead for collecting, processing, and publishing these metrics.  Since the metrics users may want is unbounded, the kubelet's resource overhead could easily grow to unreasonable levels.
+
+It is very cumbersome to make changes or bugfixes in cAdvisor, because that then needs to be vendored back into kubernetes.
+
+CAdvisor is structured to collect metrics on an interval, which is appropriate for a stand-alone metrics collector.  However, many functions in the kubelet are latency-sensitive (eviction, for example), and would benifit from a more "On-Demand" metrics collection design.
+
+### Proposal
+
+I propose to have a two seperate monitoring pipelines.
+ - The first is an internal, "Core Metrics" pipeline, which provides only those metrics consumed by kubernetes system components.
+ - The second is an external, independent "Third Party Metrics" pipeline, which provides metrics for users of kubernetes.
+
+The proposed metrics pipelines would bound the resources that the kubelet consumes, and create a seperation of concerns for kubelet metrics and user metrics.  This proposal also paves the way for simpler third party metrics integration, although that is not covered in this proposal.
+
+### Non Goals
+
+Third Party Metric Provider integration will not be tackled in this proposal. This includes both possibly providing a kubelet metadata (pod-container mappings, volume-pod mappings, etc) API, and a possible standardized metrics APIs for metrics providers to conform to.
+
+Integration with CRI will not be covered in this proposal.  In future proposals, integrating with CRI may provide a better abstraction of information required by the core metrics pipeline to collect metrics.
+
+Methods for pushing these metrics to various kubernetes components will not be covered.  This proposal does not cover any future "metrics-server", but these metrics could easily be pushed to whatever components require them.
```

Comment:
This *really*  needs to discuss it's place in the monitoring vision in more concrete terms -- if you're referencing the monitoring vision, you could just say here "this proposal does not cover exposing the metrics to components -- that is discussed in more details by the metrics-server"